src/pk.c

```diff
@@ -51,6 +51,487 @@ static int pem_write_bio_pubkey(WOLFSSL_BIO* bio, WOLFSSL_EVP_PKEY* key);
 
 #if defined(OPENSSL_EXTRA)
 
+#if (!defined(NO_DH) && (defined(HAVE_LIGHTY) || defined(HAVE_STUNNEL) || \
```

Comment:
Is there any way for us to reduce this huge gate? #infdef NO_BIO isn't sufficient?

Reply:
OPENSSL_EXTRA is defined which collapses the checks.
Changed.